src/hotspot/share/services/writeableFlags.cpp

```diff
@@ -244,6 +245,9 @@ JVMFlag::Error WriteableFlags::set_double_flag(const char* name, double value, J
 JVMFlag::Error WriteableFlags::set_ccstr_flag(const char* name, const char* value, JVMFlagOrigin origin, FormatBuffer<80>& err_msg) {
   JVMFlag* flag = JVMFlag::find_flag(name);
   JVMFlag::Error err = JVMFlagAccess::ccstrAtPut(flag, &value, origin);
+  if (err == JVMFlag::SUCCESS) {
+    assert(value == NULL, "old value is freed automatically and not returned");
+  }
```

Comment:
The whole block should be ifdef DEBUG.

Reply:
Since this whole block can be optimized out by the C compiler in product builds, I'd rather leave out the `#ifdef` to avoid clutter.